### ISSUE TITLE
AIR-Net object missing in main.ipynb

### ISSUE THREAD

**Issue by DiTo97** [lizhemin15/AIR-Net#1]
Hi,

While I was skimming through the notebook **main.ipynb** looking for a comparison between the basic DMF model and your AIR-Net model, I noticed that within the main for loop you always call the function ` demo.basic_dmf(...)`. Instead, I expected to see the `demo.air_net(...)` object being called at least once within the loop. So, how can you state that the final plot compares the NMAE of the DMF and the DMF+AIR models, if the AIR-Net model is never called in the code? Am I missing something?

From my understanding of the notebook, you compare the basic DMF model with a DMF model with two more regularization terms. Maybe the answer resides in your paper, but I haven't read it fully yet. Is the AIR-Net model any different from that?

I would l like to apply the AIR-Net model to a different matrix completion problem (unrelated to images. A sort of confusion matrix), given its promising results on paper. It would be greatly helpful if you could help me run it on your data.

**Comment by lizhemin15** [comment]
Thanks for your attention, there are some redundancy in my orginal codes, and I will update it in the next few days. And I will explain how I compare these two models in this version of codes.

You can search the following code in **main.ipynb**
`dmf = demo.basic_dmf([m,m,m,n],[reg_hc,reg_row,reg_col,reg_cnn])`
This code define four kinds of regularization term, and they are mainly useful when I test something. And the regularization which I mentioned in AIR-Net is the reg_row and reg_col.

I control the coefficient of these regularization in the following code:
`eta = [None,1e-4,1e-4,None]`
The eta is the coefficient of corresponding regularization term. If you set it as **None**, this means you do not want add this term in the loss function.

So, although I only use the demo.basic_dmf(), I also achieve the aim of using AIR-Net.

If you want to use my codes for other matrix completion problem,  you only need to adjust the following codes in **main.ipynb**,
'm,n,epoch = 240,240,10001 #Height, width and training times of the image
miss_mode = 'random' #Can be 'random', 'fixed' and 'patch'
pic = dataloader.get_data(height=n,width=n,pic_name='./train_pics/Barbara.jpg').cuda()'
Furthermore, remember to rescale your matrix elements into [0,1] or [-1,1], otherwise the algorithm may be not convergence. And adjust the coefficent of **eta**.

Thank you, I hope my answer will be helpful for you!

**Comment by DiTo97** [comment]
Thanks for the detailed answer and sorry for the late reply. I got caught up in a few busy weeks.

I will answer point for point below:

> You can search the following code in **main.ipynb** `dmf = demo.basic_dmf([m,m,m,n],[reg_hc,reg_row,reg_col,reg_cnn])` This code define four kinds of regularization term, and they are mainly useful when I test something. And the regularization which I mentioned in AIR-Net is the reg_row and reg_col.
>
> I control the coefficient of these regularization in the following code: `eta = [None,1e-4,1e-4,None]` The eta is the coefficient of corresponding regularization term. If you set it as **None**, this means you do not want add this term in the loss function.
> 
> So, although I only use the demo.basic_dmf(), I also achieve the aim of using AIR-Net.

Just to make sure I understood correctly, does this mean that calling the basic DMF function with `reg_row` and `reg_col` among its regularization parameters or calling the AIR-Net function alone achieve the very same effect?

> If you want to use my codes for other matrix completion problem, you only need to adjust the following codes in **main.ipynb**, 'm,n,epoch = 240,240,10001 #Height, width and training times of the image miss_mode = 'random' #Can be 'random', 'fixed' and 'patch' pic = dataloader.get_data(height=n,width=n,pic_name='./train_pics/Barbara.jpg').cuda()' Furthermore, remember to rescale your matrix elements into [0,1] or [-1,1], otherwise the algorithm may be not convergence. And adjust the coefficent of **eta**.

Do you think your approach can be robust to not working with images? I guess many underlying assumptions will be different, but I am trying to understand if the knowledge of the data being images is key to the success of your algorithm, or if by simply re-scaling and tuning all regularization parameters that compose **eta** accordingly to my problem I can tackle it effectively.
